### ISSUE TITLE
[unittest_vhost] AddressSanitizer: SEGV on unknown address

### ISSUE THREAD

**Issue by mikeBashStuff** [spdk/spdk#3507]
# Sighting report

`clang18` + `asan`.

```
 Test: create_controller_test ...[2024-08-26 16:30:21.104521] /root/spdk/lib/vhost/vhost.c:  81:vhost_parse_core_mask: *ERROR*: one of selected cpu is outside of core mask(=f)
[2024-08-26 16:30:21.104617] /root/spdk/lib/vhost/vhost.c: 127:vhost_dev_register: *ERROR*: cpumask 0xf0 is invalid (core mask is 0xf)
[2024-08-26 16:30:21.104714] /root/spdk/lib/vhost/vhost.c:  81:vhost_parse_core_mask: *ERROR*: one of selected cpu is outside of core mask(=f)
[2024-08-26 16:30:21.104780] /root/spdk/lib/vhost/vhost.c: 127:vhost_dev_register: *ERROR*: cpumask 0xff is invalid (core mask is 0xf)
[2024-08-26 16:30:21.104836] /root/spdk/lib/vhost/vhost.c: 121:vhost_dev_register: *ERROR*: Can't register controller with no name
AddressSanitizer:DEADLYSIGNAL
=================================================================
==215257==ERROR: AddressSanitizer: SEGV on unknown address 0x007f00665c40 (pc 0x0000004db0a0 bp 0x7fff66c9f840 sp 0x7fff66c9efc8 T0)
==215257==The signal is caused by a READ memory access.
    #0 0x4db0a0 in __sanitizer::internal_strlen(char const*) (/root/spdk/test/unit/lib/vhost/vhost.c/vhost_ut+0x4db0a0) (BuildId: d259e562114364278897d1d181b848d81f615254)
    #1 0x44f2f1 in printf_common(void*, char const*, __va_list_tag*) asan_interceptors.cpp.o
    #2 0x4503db in vasprintf (/root/spdk/test/unit/lib/vhost/vhost.c/vhost_ut+0x4503db) (BuildId: d259e562114364278897d1d181b848d81f615254)
    #3 0x631685 in spdk_vlog /root/spdk/lib/log/log.c:164:8
    #4 0x63125b in spdk_log /root/spdk/lib/log/log.c:108:2
    #5 0x537f22 in vhost_user_dev_init /root/spdk/lib/vhost/rte_vhost_user.c:1781:3
    #6 0x513abf in vhost_user_dev_create /root/spdk/lib/vhost/rte_vhost_user.c:1833:7
    #7 0x512c9b in vhost_dev_register /root/spdk/lib/vhost/vhost.c:150:8
    #8 0x5894ef in alloc_vdev /root/spdk/test/unit/lib/vhost/vhost.c/vhost_ut.c:213:7
    #9 0x557603 in create_controller_test /root/spdk/test/unit/lib/vhost/vhost.c/vhost_ut.c:411:8
    #10 0x7f8bd1bc066c  (/usr/lib64/libcunit.so.1+0x466c) (BuildId: 2b57dc493124068015ebe1e4a79a58e51353fa82)
    #11 0x7f8bd1bc09c7  (/usr/lib64/libcunit.so.1+0x49c7) (BuildId: 2b57dc493124068015ebe1e4a79a58e51353fa82)
    #12 0x7f8bd1bc1a8f in CU_run_all_tests (/usr/lib64/libcunit.so.1+0x5a8f) (BuildId: 2b57dc493124068015ebe1e4a79a58e51353fa82)
    #13 0x63962e in run_tests /root/spdk/lib/ut/ut.c:169:3
    #14 0x636a46 in spdk_ut_run_tests /root/spdk/lib/ut/ut.c:225:8
    #15 0x5549fb in main /root/spdk/test/unit/lib/vhost/vhost.c/vhost_ut.c:714:17
    #16 0x7f8bd1239087 in __libc_start_call_main (/usr/lib64/libc.so.6+0x2a087) (BuildId: 77c77fee058b19c6f001cf2cb0371ce3b8341211)
    #17 0x7f8bd123914a in __libc_start_main@GLIBC_2.2.5 (/usr/lib64/libc.so.6+0x2a14a) (BuildId: 77c77fee058b19c6f001cf2cb0371ce3b8341211)
    #18 0x42b6c4 in _start (/root/spdk/test/unit/lib/vhost/vhost.c/vhost_ut+0x42b6c4) (BuildId: d259e562114364278897d1d181b848d81f615254)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (/root/spdk/test/unit/lib/vhost/vhost.c/vhost_ut+0x4db0a0) (BuildId: d259e562114364278897d1d181b848d81f615254) in __sanitizer::internal_strlen(char const*)
==215257==ABORTING
/root/spdk/test/common/autotest_common.sh: line 1096: 215257 Aborted                 (core dumped) "$@"


Core was generated by `/root/spdk/test/unit/lib/vhost/vhost.c/vhost_ut'.
Program terminated with signal SIGABRT, Aborted.
#0  0x00007f8bd12a8664 in __pthread_kill_implementation () from /usr/lib64/libc.so.6
No symbol table info available.
#1  0x00007f8bd124fc4e in raise () from /usr/lib64/libc.so.6
No symbol table info available.
#2  0x00007f8bd1237902 in abort () from /usr/lib64/libc.so.6
No symbol table info available.
#3  0x00000000004e8afb in __sanitizer::Abort() ()
No symbol table info available.
#4  0x00000000004e6c95 in __sanitizer::Die() ()
No symbol table info available.
#5  0x00000000004cd8cf in __asan::ScopedInErrorReport::~ScopedInErrorReport() ()
No symbol table info available.
#6  0x00000000004c97ba in __asan::ReportDeadlySignal(__sanitizer::SignalContext const&) ()
No symbol table info available.
#7  0x00000000004c8ee6 in __asan::AsanOnDeadlySignal(int, void*, void*) ()
No symbol table info available.
#8  <signal handler called>
No symbol table info available.
#9  0x00000000004db0a0 in __sanitizer::internal_strlen(char const*) ()
No symbol table info available.
#10 0x000000000044f2f2 in printf_common(void*, char const*, __va_list_tag*) ()
No symbol table info available.
#11 0x00000000004503dc in __interceptor_vasprintf ()
No symbol table info available.
#12 0x0000000000631686 in spdk_vlog (level=SPDK_LOG_ERROR, file=0x652761 "/root/spdk/lib/vhost/rte_vhost_user.c", line=1782, func=0x6655c0 <__func__.vhost_user_dev_init> "vhost_user_dev_init",
    format=0x665600 <str> "Resulting socket path for controller %s is too long: %s%s\n", ap=0x7f8bcef09720) at log.c:164
        severity = 3
        buf = 0x7f8bcf30c020 "Resulting socket path for controller ", 'x' <repeats 163 times>...
        _buf = "Resulting socket path for controller ", 'x' <repeats 986 times>
        ext_buf = 0x0
        timestamp = '\000' <repeats 63 times>
        rc = 8252
#13 0x000000000063125c in spdk_log (level=SPDK_LOG_ERROR, file=0x652761 "/root/spdk/lib/vhost/rte_vhost_user.c", line=1782, func=0x6655c0 <__func__.vhost_user_dev_init> "vhost_user_dev_init",
    format=0x665600 <str> "Resulting socket path for controller %s is too long: %s%s\n") at log.c:108
        ap = {{gp_offset = 48, fp_offset = 48, overflow_arg_area = 0x7fff66ca03a0, reg_save_area = 0x7fff66ca0260}}
#14 0x0000000000537f23 in vhost_user_dev_init (vdev=0x50e000000740, name=0x7f8bcf50b060 'x' <repeats 200 times>..., cpumask=0x7f8bcf10b820, user_backend=0xc58be0 <g_vdev_user_backend>)
    at /root/spdk/lib/vhost/rte_vhost_user.c:1781
        path = "some_path/", 'x' <repeats 4085 times>
        user_dev = 0x7fff66ca0550
#15 0x0000000000513ac0 in vhost_user_dev_create (vdev=0x50e000000740, name=0x7f8bcf50b060 'x' <repeats 200 times>..., cpumask=0x7f8bcf10b820, user_backend=0xc58be0 <g_vdev_user_backend>, delay=false)
    at /root/spdk/lib/vhost/rte_vhost_user.c:1833
        rc = 1724515856
        user_dev = 0x80
#16 0x0000000000512c9c in vhost_dev_register (vdev=0x50e000000740, name=0x7f8bcf50b060 'x' <repeats 200 times>..., mask_str=0x0, params=0x0, backend=0x6de0a0 <g_vdev_backend>, user_backend=0xc58be0 <g_vdev_user_backend>,
    delay=false) at /root/spdk/lib/vhost/vhost.c:150
        cpumask = {str = '\000' <repeats 256 times>, cpus = "\017", '\000' <repeats 126 times>}
        rc = 1724516576
#17 0x00000000005894f0 in alloc_vdev (vdev_p=0x7f8bcf50b020, name=0x7f8bcf50b060 'x' <repeats 200 times>..., cpumask=0x0) at vhost_ut.c:213
        vdev = 0x50e000000740
        rc = 0
#18 0x0000000000557604 in create_controller_test () at vhost_ut.c:411
        vdev = 0x0
        vdev2 = 0x0
        ret = -22
        long_name = 'x' <repeats 4095 times>
#19 0x00007f8bd1bc066d in run_single_test.constprop () from /usr/lib64/libcunit.so.1
--Type <RET> for more, q to quit, c to continue without paging--
No symbol table info available.
#20 0x00007f8bd1bc09c8 in run_single_suite.constprop () from /usr/lib64/libcunit.so.1
No symbol table info available.
#21 0x00007f8bd1bc1a90 in CU_run_all_tests () from /usr/lib64/libcunit.so.1
No symbol table info available.
#22 0x000000000063962f in run_tests (config=0x7f8bcef09020) at ut.c:169
        suite = 0x0
        test = 0x0
#23 0x0000000000636a47 in spdk_ut_run_tests (argc=1, argv=0x7fff66ca0f38, opts=0x0) at ut.c:225
        config = {app = 0x7fff66ca19ef "/root/spdk/test/unit/lib/vhost/vhost.c/vhost_ut", test = 0x0, suite = 0x0, action = UT_ACTION_RUN_TESTS, opts = 0x0}
        rc = 0
#24 0x00000000005549fc in main (argc=1, argv=0x7fff66ca0f38) at vhost_ut.c:714
        suite = 0x5080000000a0
        num_failures = 0
(gdb)

```

**Comment by jimharris** [comment]
https://review.spdk.io/gerrit/c/spdk/spdk/+/24673

**Comment by jimharris** [comment]
Patch merged.

**Comment by mikeBashStuff** [comment]
@jimharris I think it would be beneficial to backport this fix to the v24.05.x branch.
